### PR TITLE
Examples on DatilTest + Implement Payments

### PR DESCRIPTION
**Pull request by awisniew207**
- Changed the `getPkpSessionSigs` example to not use the relayer
- All examples use `datil-test`
- All examples now mint a capacity credit NFT, create a `capacityDelegationAuthSig` and use it in the `capabilityAuthSigs` property
- Updated the `getLitActionSig` example to generate a `litActionCode` hash locally and add it as a permitted action to the PKP
- Updated the README.md files with the relevant information

### CODE REVIEW

File: session-signatures/getLitActionSessionSigs/README.md [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-2. Connect to the Lit network using the `LitNodeClient` on the `datil-dev` network, specifying the local storage to hold the generated wallet and session keys
+2. Connect to the Lit network using the `LitNodeClient` on the `datil-test` network, specifying the local storage to hold the generated session signature and keypair
 3. Connect the `LitContracts` client to the Lit network
-4. Mint a PKP using the `pkpNftContractUtils.write.mint()` method from `LitContracts`.
+4. Mint a PKP using the `pkpNftContractUtils.write.mint` method from `LitContracts`.
     - You may receive a gas error if your `ethersSigner` has insufficient tokens. This can be fixed by acquiring more tokens from [the faucet](https://chronicle-yellowstone-faucet.getlit.dev/) for the `Chronicle Yellowstone` blockchain
-5. Get the session signatures for the newly minted PKP, specifying the ability to execute `Lit Actions`
+5. Mint a [`capacityCreditsNFT`](https://developer.litprotocol.com/sdk/capacity-credits) and define the request limit and expiration date
+6. Use the `capacityCreditsNFT` to create a `capacityDelegationAuthSig`
+7. Define the `litActionCode` and locally create a hash of it
```
Comment (spacesailor24) [inline session-signatures/getLitActionSessionSigs/README.md:13]:
```suggestion
7. Define the `litActionCode` and locally create the IPFS CID hash of it
```

File: session-signatures/getLitActionSessionSigs/src/index.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ import {
 } from "@lit-protocol/auth-helpers";
 import * as ethers from "ethers";
 import { LocalStorage } from "node-localstorage";
+// @ts-ignore
```
Comment (spacesailor24) [inline session-signatures/getLitActionSessionSigs/src/index.ts:11]: What was the error that needed to be ignored?

File: session-signatures/getLitActionSessionSigs/src/index.ts [hunk 3]
```diff
@@ -48,20 +50,40 @@ export const getSessionSigsLitAction = async () => {
...
     );
 
+    console.log("🔄 Minting Capacity Credits NFT...");
+    const capacityTokenId = (
+      await litContracts.mintCapacityCreditsNFT({
+        requestsPerKilosecond: 10,
+        daysUntilUTCMidnightExpiration: 1,
+      })
+    ).capacityTokenIdStr;
+    console.log(`✅ Minted new Capacity Credit with ID: ${capacityTokenId}`);
```
Comment (spacesailor24) [inline session-signatures/getLitActionSessionSigs/src/index.ts:60]: I think the `getSessionSigsLitAction` function should take a Capacity Credit as an argument, and only mint one if one isn't provided - what do you think? There's no need to mint a credit every single time

File: session-signatures/getLitActionSessionSigs/src/index.ts [hunk 4]
```diff
@@ -48,20 +50,40 @@ export const getSessionSigsLitAction = async () => {
...
+      await litNodeClient.createCapacityDelegationAuthSig({
+        dAppOwnerWallet: ethersSigner,
+        capacityTokenId,
+        delegateeAddresses: [pkp.ethAddress],
+        uses: "1",
+      });
+    console.log(`✅ Created the capacityDelegationAuthSig`);
+
     console.log("🔄 Adding example permitted Lit Action to the PKP");
+    const litActionCode = `(async () => {LitActions.setResponse({ response: makeItTrue });})();`;
```
Comment (spacesailor24) [inline session-signatures/getLitActionSessionSigs/src/index.ts:73]: If we want to showcase passing a param to the Lit Action code, then I think we should do something different:

```
(() => {
    if (magicNumber >= 42) {
        LitActions.setResponse({ response:"true" });
    } else {
        LitActions.setResponse({ response: "false" });
    }
})();

File: session-signatures/getPkpSessionSigs/README.md [hunk 5]
```diff
@@ -1,19 +1,22 @@
...
+5. Mint a [`capacityCreditsNFT`](https://developer.litprotocol.com/sdk/capacity-credits) and define the request limit and expiration date
+6. Use the `capacityCreditsNFT` to create a `capacityDelegationAuthSig`
+7. Generate a wallet signature to use as an [`AuthMethod`](https://v6-api-doc-lit-js-sdk.vercel.app/interfaces/types_src.AuthMethod.html)
+8. Get the session signatures for the newly minted PKP. Any network costs will be undertaken by the `dAppOwnerWallet` specified in the `capacityDelegationAuthSig`
 
 **NOTE**
 ---
 
-When using `getPkpSessionSigs()`, you have multiple authentication options. You can use `AuthSigs`, `AuthMethods`, or `Lit Actions`, and you're not strictly limited to using an `AuthMethod`.
+When using `getPkpSessionSigs`, you have multiple authentication options. You can use `AuthSigs`, `AuthMethods`, or `Lit Actions`, and you're not strictly limited to using an `AuthMethod`.
```
Comment (spacesailor24) [inline session-signatures/getPkpSessionSigs/README.md:19]:
```suggestion
When using `getPkpSessionSigs`, you have multiple authentication options. You're not strictly limited to using an `AuthMethod`, you can also use `AuthSigs`, or `Lit Actions`.
```

File: session-signatures/getLitActionSessionSigs/src/index.ts (hunk 2 above)
Comment (awisniew207) [reply session-signatures/getLitActionSessionSigs/src/index.ts:11]: Could not find a declaration file for module 'ipfs-only-hash'.

File: session-signatures/getLitActionSessionSigs/src/index.ts (hunk 3 above)
Comment (spacesailor24) [reply session-signatures/getLitActionSessionSigs/src/index.ts:60]: If we do implement this, we should probably do the same for PKPs? We just don't want users thinking they **have** to mint new credits/PKPs

Comment (awisniew207) [reply session-signatures/getLitActionSessionSigs/src/index.ts:60]: Alternatively we could have users provide the `capacityTokenIdStr` in the `.env` file. Thoughts?

File: session-signatures/getLitActionSessionSigs/src/index.ts (hunk 4 above)
Comment (awisniew207) [reply session-signatures/getLitActionSessionSigs/src/index.ts:73]: Changed this. Just wanted to confirm, even when the magicNumber is less than 42 should sessionsigs still be generated? (they are)

File: session-signatures/getLitActionSessionSigs/src/index.ts (hunk 2 above)
Comment (spacesailor24) [reply session-signatures/getLitActionSessionSigs/src/index.ts:11]: Looks like we can use https://www.npmjs.com/package/typestub-ipfs-only-hash

File: session-signatures/getLitActionSessionSigs/src/index.ts (hunk 3 above)
Comment (spacesailor24) [reply session-signatures/getLitActionSessionSigs/src/index.ts:60]: Well passing it as an argument makes it easier to write the test files. For the tests we want to mint new ones and use existing ones, and that's harder to do if it's read from `.env`

File: session-signatures/getLitActionSessionSigs/src/index.ts (hunk 4 above)
Comment (spacesailor24) [reply session-signatures/getLitActionSessionSigs/src/index.ts:73]: No, they shouldn't, not sure why they are

Comment (awisniew207) [reply session-signatures/getLitActionSessionSigs/src/index.ts:73]: Works!

**Review by spacesailor24 — APPROVED** [review]

**Pull request merged**